### DISCUSSION TITLE
Why Nominatim stores multiple row of same OSM Object?

### DISCUSSION THREAD

**Discussion by yukha-dw** [osm-search/Nominatim#3249] [Q&A] [answered]
As title said. It's strange to see identic OSM Object with different `place_id` and slightly different `class && type` values. Is there any advantage of doing this?

Here's the example of duplicated OSM Object. I'm not quite sure if [sort_and_cut_results](https://github.com/osm-search/Nominatim/blob/6478409b05c0f72f5113a8c30819b81cc9641be3/nominatim/api/search/geocoder.py#L106-L123) function will always drop the duplicates but it happens in this case:
https://nominatim.openstreetmap.org/search.php?q=%EC%84%9C%EB%9E%98%EC%84%AC&format=jsonv2&debug=1

**Query on Postgres**

```SQL
SELECT
 place_id,importance, rank_address, rank_search, osm_type, osm_id, class, type, admin_level, cast(name as json),
  cast(address as json), cast(extratags as json), wikipedia, country_code,
 housenumber, postcode, cast(ST_AsGeoJSON(centroid) as json) as centroid
FROM
 placex
WHERE
 osm_id = '414891746';
```

| place_id | importance | rank_address | rank_search | osm_type | osm_id | class | type | admin_level | name | address | extratags | wikipedia | country_code | housenumber | postcode | centroid |
|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|
|626443|0.25000999999999995 | 0 | 20 | W | 414891746 | place | islet | 15 | {"name": "서래섬", "name:en": "Seorae Island", "name:ja": "ソレ島", "name:ko": "서래섬"} | | | | kr | | | {"type":"Point","coordinates":[126.990247212,37.50816095]}|
|626471|0.15000999999999998 | 0 | 24 | W | 414891746 | leisure | park | 15 | {"name": "서래섬", "name:en": "Seorae Island", "name:ja": "ソレ島", "name:ko": "서래섬"} | | {"place": "islet"} | | kr | | | {"type":"Point","coordinates":[126.990247212,37.50816095]}|

**Comment by lonvia** [comment, marked as answer]
This is an OSM object that serves two different primary functions. Nominatim duplicates them to be sure they can be found with both functions.
